misc/quitcd/quitcd.nu

```
@@ -1,19 +1,27 @@
-# The behaviour is set to cd on quit (nnn checks if NNN_TMPFILE is set)
-let cfgHome = ($env | default $"($env.HOME)/.config" XDG_CONFIG_HOME | get XDG_CONFIG_HOME)
-$env.NNN_TMPFILE = $"($cfgHome)/nnn/.lastd"
+# Run nnn with dynamic changing directory to the environment.
+export def --env n [
+	...flags : string # Extra flags to launch nnn with.
+] -> nothing {
 
-def --env n [...x] {
-  # Launch nnn. Add desired flags after `^nnn`, ex: `^nnn -eda ($x | str join)`
-  ^nnn ($x | str join)
-  let newpath = (
-    if ($env.NNN_TMPFILE | path exists) {
-      # FIXME: fails if path contains single-quote
-      let newpath = (open $env.NNN_TMPFILE | parse "cd '{nnnpath}'").0.nnnpath
-      ^rm -f $env.NNN_TMPFILE
-      echo $newpath
-    } else {
-      pwd
-    }
-  )
-  cd $newpath
+	let config_home = if ($env.XDG_CONFIG_HOME | is-empty) {
+		$'($env.HOME)/.config/'
+	} else {
+		$env.XDG_CONFIG_HOME
+	}
+
+	# The behaviour is set to cd on quit (nnn checks if NNN_TMPFILE is set).
+	let nnn_tmpfile = $config_home + '/nnn/.lastd'
+
+	# Launch nnn. Add desired flags after `^nnn`, ex: `^nnn -eda ($x | str join)`
+	NNN_TMPFILE=$nnn_tmpfile ^nnn ...$flags
+
+	if ($nnn_tmpfile | path exists) {
+		# Remove <cd '> from the first part of the string and the last single quote <'>.
+		# Fix post-processing of nnn with path, escaping single quotes with POSIX syntax.
+		let path = (open $nnn_tmpfile | str substring 4..-1 | str replace --all `'\''` `'`)
+
+		rm $nnn_tmpfile
```

without the --force option in  `rm $nnn_tmpfile` I have now a confirmation prompt when I exit nnn : `rm: remove '/home/henri/.config/nnn/.lastd'?  [Y/N]:`

---

Can't replicate, which version are you running? If the same, seems like a **nushell** platform's incompatibility issue assuming that you don't use Linux, given `rm` information, the force flag should only regard itself to ignore errors, not make it non-interactive like the **GNU coreutils** `rm`:

```
rm (Nushell 0.89.0)
...
  -f, --force - suppress error when no file
...
  Delete a file, ignoring 'file not found' errors
  > rm --force file.txt
```

But assuming also that that file can be deleted before it runs that line, adding the force flag may not cause any harm. If we're on the same version, I can add the flag and, hopefully, review it later.

---

I use nushell 0.87 on NixOS.

The confirmation prompt occurs when you have an alias  `alias rm = rm -i`  in your nushell config (which is common).
You are right the "--force" option on the nushell `rm` command don't change anything, you have to use the external `rm`  with the caret prefix :   `^rm -f`.

---

Try `^rm` without the force flag.

---

Yes. `^rm $nnn_tmpfile` works.